### PR TITLE
Define the patch locale of a spectral locale

### PR DESCRIPTION
**Pull request by ayberkt**
The long-overdue construction of the patch locale is finally here.

### CODE REVIEW

File: source/CompactRegular.lagda [hunk 1]
```diff
@@ -793,3 +793,187 @@ clopen-iff-compact-in-stone-frame F (κ , ζ) U = β , γ
...
+
+     ii : ((⋁[ F ] ⁅ ℬ [ i ] ∣ i ε ℐ ⁆) ≤[ poset-of F ] V) holds
+     ii = ⋁[ F ]-least ⁅ ℬ [ i ] ∣ i ε ℐ ⁆ (V , ξ)
+
+\end{code}
+
+\begin{code}
+
+compacts-are-basic-in-spectral-frames : (F : frame 𝓤 𝓥 𝓦)
+                                      → (σ : is-spectral₀ F)
```
Comment (martinescardo) [inline source/CompactRegular.lagda:910]: Rename "is-spectral₀" to "spectral-structure".

Comment (martinescardo) [reply source/CompactRegular.lagda:910]: Or "spectral-data" maybe is better.

Comment (martinescardo) [reply source/CompactRegular.lagda:910]: Rename "compacts-are-basic-in-spectral-frames" to "compacts-are-basic-in-spectralᴰ-frames".

Whenever you work with spectrality as data, use a "D" superscript as above.

Comment (martinescardo) [reply source/CompactRegular.lagda:910]: More generally, whenever you work with the data version of something, use a "D" superscript.

File: source/CompactRegular.lagda [hunk 2]
```diff
@@ -793,3 +793,187 @@ clopen-iff-compact-in-stone-frame F (κ , ζ) U = β , γ
...
+     → ∥ Σ i ꞉ index ℬ↑ , U ≡ ℬ↑ [ i ] ∥
+   γ (k , p) = ∣ ℐ [ k ] , ≤-is-antisymmetric (poset-of F) p β ∣
+    where
+     β : ((ℬ↑ [ ℐ [ k ] ]) ≤[ poset-of F ] U) holds
+     β = ℬ↑ [ ℐ [ k ] ]              ≤⟨ ⋁[ F ]-upper ⁅ ℬ↑ [ i ] ∣ i ε ℐ ⁆ k ⟩
+         ⋁[ F ] ⁅ ℬ↑ [ i ] ∣ i ε ℐ ⁆ ≡⟨ covers F ℬ↑ b↑ U ⁻¹                 ⟩ₚ
+         U                           ■
+
+-- TODO: it's not clear if this lemma will be needed. Think more about this and
+-- remove it if it turns out that it won't be needed.
```
Comment (martinescardo) [inline source/CompactRegular.lagda:954]: Keep it anyway.

File: source/Nucleus.lagda [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+\begin{code}
+
+is-inflationary : (L : frame 𝓤 𝓥 𝓦) → (⟨ L ⟩ → ⟨ L ⟩) → Ω (𝓤 ⊔ 𝓥)
+is-inflationary L j = Ɐ x ∶ ⟨ L ⟩ , x ≤[ poset-of L ] j x
+
+is-idempotent : (L : frame 𝓤 𝓥 𝓦) → (⟨ L ⟩ → ⟨ L ⟩) → Ω (𝓤 ⊔ 𝓥)
+is-idempotent L j = Ɐ x ∶ ⟨ L ⟩ , j (j x) ≤[ poset-of L ] j x
+
+is-nuclear : (L : frame 𝓤 𝓥 𝓦) → (⟨ L ⟩ → ⟨ L ⟩) → Ω (𝓤 ⊔ 𝓥)
```
Comment (martinescardo) [inline source/Nucleus.lagda:38]: Rename "is-nuclear" to "is-nucleus".

Comment (martinescardo) [reply source/Nucleus.lagda:38]: Shouldn't "preserves-meets" be called "preserves-finite-meets"? Or "preserves-binary-meets"?

File: source/Nucleus.lagda [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+\end{code}
+
+In this development, it will be useful to define and work with the notion of a
+prenucleus: a meet-preserving inflationary endomap (that is not necessary
+idempotent):
+
+\begin{code}
+
+is-prenuclear : (L : frame 𝓤 𝓥 𝓦) (j : ⟨ L ⟩ → ⟨ L ⟩) → Ω (𝓤 ⊔ 𝓥)
```
Comment (martinescardo) [inline source/Nucleus.lagda:97]: Rename "is-prenuclear" to "is-prenucleus".

File: source/PatchTopology.lagda [hunk 5]
```diff
@@ -0,0 +1,886 @@
...
+\begin{code}
+
+ is-perfect : (⟨ 𝒪 X ⟩ → ⟨ 𝒪 X ⟩) → Ω (𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺)
+ is-perfect = is-scott-continuous (𝒪 X) (𝒪 X)
+
+\end{code}
+
+\begin{code}
+
+ perfect-nucleus : 𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺ ̇
```
Comment (martinescardo) [inline source/PatchTopology.lagda:85]: A convention is that type names are capital. So use "Perfect-Nucleus".

File: source/PatchTopology.lagda [hunk 6]
```diff
@@ -0,0 +1,886 @@
...
+```
+S ∶≡ { jᵢ : 𝒪 X → 𝒪 X ∣ i ∶ I }
+```
+
+of prenuclei, `sequence S is` is a prenuclei for any given list `is : List I` of
+indices.
+
+\begin{code}
+
+ 𝔡𝔦𝔯-prenuclear : (K : Fam 𝓦 (⟨ 𝒪 X ⟩ → ⟨ 𝒪 X ⟩))
```
Comment (martinescardo) [inline source/PatchTopology.lagda:378]: "𝔡𝔦𝔯-prenuclear" is a terrible name. :-)
What about "family-of-finite-compositions"?

File: source/PatchTopology.lagda [hunk 7]
```diff
@@ -0,0 +1,886 @@
...
+      † (j ∷ js) U = (𝔡𝔦𝔯 K₀ [ js ]) ((K₀ [ j ]) U)  ≤⟨ ♠           ⟩
+                     (𝔡𝔦𝔯 K₀ [ js ]) (k U)           ≤⟨ † js (k U)  ⟩
+                     k (k U)                         ≤⟨ n₂ U        ⟩
+                     k U                             ■
+                      where
+                       ♠ = prenuclei-are-monotone (𝒪 X) (K₁ ^* [ js ]) _ (φ j U)
+
+\end{code}
+
+It's hard to find a good name for the following two lemmas, which are crucial
```
Comment (martinescardo) [inline source/PatchTopology.lagda:718]: It's fine to use obscure names for technical lemmas.

**Review by martinescardo — CHANGES_REQUESTED** [review]

File: source/Nucleus.lagda (hunk 4 above)
Comment (ayberkt) [reply source/Nucleus.lagda:97]: Done.

**Pull request merged**